ibm/resource_ibm_ob_logging.go

```diff
@@ -0,0 +1,358 @@
+
+	client, err := meta.(ClientSession).VpcContainerAPI()
+	if err != nil {
+		return err
+	}
+
+	parts, err := idParts(d.Id())
+	if err != nil {
+		return err
+	}
+	clusterName := parts[0]
+	loggingID := parts[1]
+
+	targetEnv, err := getLoggingTargetHeader(d, meta)
+	if err != nil {
+		return err
+	}
+
+	config, err := client.Logging().GetLoggingConfig(clusterName, loggingID, targetEnv)
+	if err != nil {
+		return err
+	}
+
+	d.Set(obLoggingPrivateEndpoint, config.PrivateEndpoint)
+	d.Set(obLoggingDaemonSetName, config.DaemonsetName)
+	d.Set(obLoggingInstanceName, config.InstanceName)
+	d.Set(obLoggingAgentKey, config.AgentKey)
+	d.Set(obLoggingAgentNamespace, config.AgentNamespace)
+	d.Set(obLoggingDiscoveredAgent, config.DiscoveredAgent)
+	d.Set(obLoggingCrn, config.CRN)
+	d.Set(obLoggingNamespace, config.Namespace)
+
+	return nil
+
+}
+
+func resourceIBMLoggingUpdate(d *schema.ResourceData, meta interface{}) error {
+
+	hasChanged := false
+	idChanged := false
+
+	client, err := meta.(ClientSession).VpcContainerAPI()
+	if err != nil {
+		return err
+	}
+
+	targetEnv, err := getLoggingTargetHeader(d, meta)
+	if err != nil {
+		return err
+	}
+
+	loggingUpdateModel := v2.LoggingUpdateRequest{}
+
+	parts, err := idParts(d.Id())
+	if err != nil {
+		return err
+	}
+	cluster := parts[0]
+	loggingID := parts[1]
+
+	clusterName := cluster
+	loggingUpdateModel.Cluster = clusterName
+
+	if d.HasChange(obLoggingInstanceID) {
+		hasChanged = true
+		idChanged = true
+		old, new := d.GetChange(obLoggingInstanceID)
+		loggingUpdateModel.Instance = old.(string)
+		loggingUpdateModel.NewInstance = new.(string)
+	} else {
+		loggingUpdateModel.Instance = loggingID
+	}
+
+	if d.HasChange(obLoggingIngestionkey) {
+		key := d.Get(obLoggingIngestionkey).(string)
+		loggingUpdateModel.IngestionKey = key
+		hasChanged = true
+	}
+
+	if d.HasChange(obLoggingPrivateEndpoint) {
+		endpoint := d.Get(obLoggingPrivateEndpoint).(bool)
+		loggingUpdateModel.PrivateEndpoint = endpoint
+		hasChanged = true
+	}
+
+	if hasChanged {
+
+		_, err := client.Logging().UpdateLoggingConfig(loggingUpdateModel, targetEnv)
+		if err != nil {
+			return err
+		} else if idChanged {
+			d.SetId(fmt.Sprintf("%s/%s", clusterName, loggingUpdateModel.NewInstance))
+		}
+	}
+
+	return resourceIBMLoggingRead(d, meta)
+}
+
+func resourceIBMLoggingDelete(d *schema.ResourceData, meta interface{}) error {
+	client, err := meta.(ClientSession).VpcContainerAPI()
+	if err != nil {
+		return err
+	}
+
+	targetEnv, err := getLoggingTargetHeader(d, meta)
+	if err != nil {
+		return err
+	}
+
+	parts, err := idParts(d.Id())
+	if err != nil {
+		return err
+	}
+	clusterName := parts[0]
+	loggingID := parts[1]
+
+	//populate logging logDNA configure create request
+	params := v2.LoggingDeleteRequest{
+		Cluster:  clusterName,
+		Instance: loggingID,
+	}
+
+	_, err = client.Logging().DeleteLoggingConfig(params, targetEnv)
+	if err != nil {
```

Comment:
same here 
Inside the err check if resp.statuscode is 404 then set d.SetID("") and return nil
else retrun err